programs/ssl/ssl_client2.c

```diff
@@ -1971,7 +1965,21 @@ int main(int argc, char *argv[])
     }
 
 #if defined(MBEDTLS_SSL_EARLY_DATA)
-    mbedtls_ssl_conf_early_data(&conf, opt.early_data);
+    int early_data_enabled = MBEDTLS_SSL_EARLY_DATA_DISABLED;
+    FILE *early_data_fp = NULL;
+    size_t early_data_len = 0;
+    if (strlen(opt.early_data_file) > 0) {
+        if ((early_data_fp = fopen(opt.early_data_file, "rb")) == NULL) {
+            mbedtls_printf("failed\n  ! Cannot open '%s' for reading.\n",
+                           opt.early_data_file);
+            goto exit;
+        }
+        early_data_len = fread(buf, 1, sizeof(buf), early_data_fp);
```

Comment:
Nitpick comment, not necessary for this PR
```suggestion
```

Reply:
Actually, this is related with the design we talked about in https://github.com/Mbed-TLS/mbedtls/pull/8596#discussion_r1416618029
We want a input file which isn't empty.
Anyway, it depend on the result of the discussion.